janus.go

```diff
@@ -243,37 +464,76 @@ func (gateway *Gateway) Create() (*Session, error) {
 	// Create new session
 	session := new(Session)
 	session.gateway = gateway
-	session.ID = success.Data.ID
+	session.id = success.Data.ID
 	session.Handles = make(map[uint64]*Handle)
-	session.Events = make(chan interface{}, 2)
+	session.events = make(chan interface{}, 20)
+	session.CloseCh = make(chan bool)
+	session.CloseChLock = &sync.Mutex{}
 
 	// Store this session
 	gateway.Lock()
-	gateway.Sessions[session.ID] = session
+	gateway.Sessions[session.id] = session
 	gateway.Unlock()
 
 	return session, nil
 }
 
 // Session represents a session instance on the Janus Gateway.
 type Session struct {
-	// ID is the session_id of this session
-	ID uint64
+	// id is the session_id of this session
+	id uint64
 
 	// Handles is a map of plugin handles within this session
 	Handles map[uint64]*Handle
 
-	Events chan interface{}
+	events chan interface{}
 
 	// Access to the Handles map should be synchronized with the Session.Lock()
 	// and Session.Unlock() methods provided by the embeded sync.Mutex.
 	sync.Mutex
 
-	gateway *Gateway
+	CloseCh     chan bool
+	CloseChLock *sync.Mutex
+	gateway     *Gateway
+}
+
+//LongPoll ...
+func (session *Session) LongPoll() {
+	defer func() {
+	}()
```

Comment:
?

Reply:
oops that had a log line in it, was supposed to be removed